### PR TITLE
Add additional documentation for setting an owner on contract deployment

### LINKED ISSUE #148: Update Ownership library documentation for setting ownership on deployment
#### Description

You can now set ownership on deployment without needing to call the `set_ownership()` or a constructor function. To set the ownership on deployment, add the address to the storage block:

```rust
storage {
    owner: Ownership = Ownership::initalized(Identity::Address(Address::from(0x0000000000000000000000000000000000000000000000000000000000000000))),
}
```

#### Motivation

This is shown in the inline documentation but not documented properly in the README.

### PR DESCRIPTION
**Pull request by bitzoic**
## Type of change

- Documentation

## Changes

The following changes have been made:

- Added documentation outlining how to initialize ownership on contract deployment.
- Added new functions to SPECIFICATION
- Added missing Arguments inline documentation
- Re-organized inline documentation 

## Related Issues

Closes #148

### CODE REVIEW

File: libs/ownership/SPECIFICATION.md [hunk 1]
```diff
@@ -24,8 +24,22 @@ Only callable by the current owner, this function will remove the owner.
 
 ### `set_ownership()`
 
-This function will store a new owner upon initalization.
+This function will store a new owner if there has never been one set.
```
Comment (Braqzen) [inline libs/ownership/SPECIFICATION.md:27]:
```suggestion
This function will store a new owner if one has not been set.
```

File: libs/ownership/README.md [hunk 2]
```diff
@@ -9,50 +9,57 @@
...
 
-You may import items like so:
+Once imported, the `Ownership` struct should be added to the storage block of your contract. There are two approaches when declaring ownership in storage.
+
+1. Initalize the owner on contract deployment by calling the `initalized()` function.
 
 ```rust
-use ownership::ownable::{only_owner, owner, set_ownership};
+storage {
+    owner: Ownership = Ownership::initalized(Identity::Address(Address::from(0x0000000000000000000000000000000000000000000000000000000000000000))),
```
Comment (Braqzen) [inline libs/ownership/README.md:32]:
```suggestion
    owner: Ownership = Ownership::initialized(Identity::Address(Address::from(0x0000000000000000000000000000000000000000000000000000000000000000))),
```

File: libs/ownership/README.md [hunk 3]
```diff
@@ -9,50 +9,57 @@
...
-```toml
-ownership = { git = "https://github.com/fuellabs/sway-libs", branch = "master" }
+```rust
+use ownership::Ownership;
 ```
 
-You may import items like so:
+Once imported, the `Ownership` struct should be added to the storage block of your contract. There are two approaches when declaring ownership in storage.
+
+1. Initalize the owner on contract deployment by calling the `initalized()` function.
```
Comment (Braqzen) [inline libs/ownership/README.md:28]:
```suggestion
1. Initialize the owner on contract deployment by calling the `initialized()` function.
```

File: libs/ownership/README.md [hunk 4]
```diff
@@ -9,50 +9,57 @@
...
 
 ```rust
-use ownership::ownable::{only_owner, owner, set_ownership};
+storage {
+    owner: Ownership = Ownership::initalized(Identity::Address(Address::from(0x0000000000000000000000000000000000000000000000000000000000000000))),
+}
 ```
 
-Once imported, an owner can be set by calling the `set_ownership` function. 
+2. Leave the owner uninitialized and call the `set_ownership()` function in your own constructor.
```
Comment (Braqzen) [inline libs/ownership/README.md:36]:
```suggestion
2. Leave the owner uninitialized and call the `set_ownership()` function in your own constructor.
```

File: libs/ownership/SPECIFICATION.md [hunk 5]
```diff
@@ -24,8 +24,22 @@ Only callable by the current owner, this function will remove the owner.
...
 
 ### `transfer_ownership()`
 
 Only callable by the current owner, this function will transfer ownership to another user.
+
+### `uninitialized()`
+
+Creates a new ownership in the `Uninitalized` state.
+
+### `initalized()`
```
Comment (Braqzen) [inline libs/ownership/SPECIFICATION.md:37]:
```suggestion
### `initialized()`
```

File: libs/ownership/SPECIFICATION.md [hunk 6]
```diff
@@ -24,8 +24,22 @@ Only callable by the current owner, this function will remove the owner.
...
+
+Creates a new ownership in the `Initalized` state.
```
Comment (Braqzen) [inline libs/ownership/SPECIFICATION.md:39]:
```suggestion
Creates a new ownership in the `Initialized` state.
```

File: libs/ownership/SPECIFICATION.md [hunk 7]
```diff
@@ -24,8 +24,22 @@ Only callable by the current owner, this function will remove the owner.
...
+
+### `revoked()`
+
+Creates a new ownership in the `Revkoed` state.
```
Comment (Braqzen) [inline libs/ownership/SPECIFICATION.md:43]:
```suggestion
Creates a new ownership in the `Revoked` state.
```

File: libs/ownership/SPECIFICATION.md [hunk 8]
```diff
@@ -24,8 +24,22 @@ Only callable by the current owner, this function will remove the owner.
...
+
+> **Note** Once the ownership has been revoked it cannot be set or transfered again.
```
Comment (Braqzen) [inline libs/ownership/SPECIFICATION.md:45]:
```suggestion
> **Note** Once the ownership has been revoked it cannot be set or transferred again.
```

File: libs/ownership/SPECIFICATION.md [hunk 9]
```diff
@@ -24,8 +24,22 @@ Only callable by the current owner, this function will remove the owner.
...
-This function will store a new owner upon initalization.
+This function will store a new owner if there has never been one set.
 
 ### `transfer_ownership()`
 
 Only callable by the current owner, this function will transfer ownership to another user.
+
+### `uninitialized()`
+
+Creates a new ownership in the `Uninitalized` state.
```
Comment (Braqzen) [inline libs/ownership/SPECIFICATION.md:35]:
```suggestion
Creates a new ownership in the `Uninitialized` state.
```

File: libs/ownership/README.md (hunk 4 above)
Comment (bitzoic) [reply libs/ownership/README.md:36]: There is nothing different here

Comment (Braqzen) [reply libs/ownership/README.md:36]: Yh, I resolved it for that reason. I refreshed partly through review when github had a popup button so maybe something changed to be the same

**Review by K1-R1 — APPROVED** [review]

**Comment by K1-R1** [comment]
Too slow

**Pull request merged**